pkg/core/dao/dao.go

```diff
@@ -34,11 +35,33 @@ var (
 type Simple struct {
 	Version Version
 	Store   *storage.MemCachedStore
+
+	nativeCacheLock sync.RWMutex
+	nativeCache     map[int32]NativeContractCache
+	nativeCachePS   NativeContractCacheStore
+
 	private bool
 	keyBuf  []byte
 	dataBuf *io.BufBinWriter
 }
 
+// NativeContractCacheStore is an interface that provides functionality to store
+// and retrieve multi-tier native contract cache.
+type NativeContractCacheStore interface {
```

Comment:
Why do you need this interface?

Reply:
We can safely go without it. But we don't need the whole DAO's functionality to be presented inside the `nativeCachePS`, we only need those functionality that is related to native cache storing.

Should we remove it?

Reply:
Fixed.